scipy/stats/_stats_py.py

```diff
@@ -5202,72 +5202,163 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate',
+    distribution for large samples is the Student's t distribution with
+    ``len(x) - 2`` degrees of freedom.
+
+    >>> import matplotlib.pyplot as plt
+    >>> dof = len(x)-2  # len(x) == len(y)
+    >>> dist = stats.t(df=dof)
+    >>> t_vals = np.linspace(-5, 5, 100)
+    >>> pdf = dist.pdf(t_vals)
+    >>> fig, ax = plt.subplots(figsize=(8, 5))
+    >>> def plot(ax):  # we'll re-use this
+    ...     ax.plot(t_vals, pdf)
+    ...     ax.set_title("Spearman's Rho Test Null Distribution")
+    ...     ax.set_xlabel("statistic")
+    ...     ax.set_ylabel("probability density")
+    >>> plot(ax)
+    >>> plt.show()
+
+    The comparison is quantified by the p-value: the proportion of values in
+    the null distribution as extreme or more extreme than the observed
+    value of the statistic. In a two-sided test in which the statistic is
+    positive, elements of the null distribution greater than the transformed
+    statistic and elements of the null distribution less than the negative of
+    the observed statistic are both considered "more extreme".
+
+    >>> fig, ax = plt.subplots(figsize=(8, 5))
+    >>> plot(ax)
+    >>> rs = res.statistic  # original statistic
+    >>> transformed = rs * np.sqrt(dof / ((rs+1.0)*(1.0-rs)))
+    >>> pvalue = dist.cdf(-transformed) + dist.sf(transformed)
+    >>> annotation = (f'p-value={pvalue:.4f}\n(shaded area)')
+    >>> props = dict(facecolor='black', width=1, headwidth=5, headlength=8)
+    >>> _ = ax.annotate(annotation, (2.7, 0.025), (3, 0.03), arrowprops=props)
+    >>> i = t_vals >= transformed
+    >>> ax.fill_between(t_vals[i], y1=0, y2=pdf[i], color='C0')
+    >>> i = t_vals <= -transformed
+    >>> ax.fill_between(t_vals[i], y1=0, y2=pdf[i], color='C0')
+    >>> ax.set_xlim(-5, 5)
+    >>> ax.set_ylim(0, 0.1)
+    >>> plt.show()
     >>> res.pvalue
-    0.08858700531354381
-    >>> rng = np.random.default_rng()
-    >>> x2n = rng.standard_normal((100, 2))
-    >>> y2n = rng.standard_normal((100, 2))
-    >>> res = stats.spearmanr(x2n)
-    >>> res.statistic, res.pvalue
-    (-0.07960396039603959, 0.4311168705769747)
-    >>> res = stats.spearmanr(x2n[:, 0], x2n[:, 1])
-    >>> res.statistic, res.pvalue
-    (-0.07960396039603959, 0.4311168705769747)
-    >>> res = stats.spearmanr(x2n, y2n)
+    0.07991669030889909
+
+    If the p-value is "small" - that is, if there is a low probability of
+    sampling data from independent distributions that produces such an extreme
+    value of the statistic - this may be taken as evidence against the null
+    hypothesis in favor of the alternative: the distribution of total collagen
+    and free proline are *not* independent. Note that:
+
+    - The inverse is not true; that is, the test is not used to provide
+      evidence for the null hypothesis.
+    - The threshold for values that will be considered "small" is a choice that
+      should be made before the data is analyzed [3]_ with consideration of the
```

Comment:
Not reference 3 here. Check this.

Reply:
What do you mean?

Reply:
That was a note to self. I need to fix the reference number.